core/controllers/feedback_test.py

```diff
@@ -554,3 +556,216 @@ def test_feedback_threads_with_suggestions(self):
         }
         self.assertDictContainsSubset(
             expected_suggestion_dict, response['suggestion'])
+
+    def test_get_feedback_threads_with_no_text_and_no_updated_status_raises_400(
+            self):
+        self.login(self.OWNER_EMAIL_1)
+        response = self.get_html_response('/create/%s' % self.EXP_ID)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        feedback_services.create_thread(
+            feconf.ENTITY_TYPE_EXPLORATION, self.EXP_ID, self.owner_id_1,
+            'a subject', 'some text', has_suggestion=True)
```

Comment:
I would suggest setting has_suggestion=False (and just test the flows with feedback threads, for all of the tests in this PR). The suggestion framework is tested separately, and I don't think these tests need to test that behavior. Let me know your thoughts.

Reply:
+1, good idea!

Reply:
But doesn't the handler `ThreadHandler` works with suggestions? I think the thread id there is interpreted as suggestion_id

Reply:
It works with both... So there are 2 types, feedback threads and suggestion threads. This handler works with threads (of both types). You can create feedback threads without having a suggestion (e.g. if you just leave a feedback comment at the end of the exploration). For the purpose of all tests here, you can just create feedback threads IMO. No need to create suggestion threads. I know that the threadListHandler has some checks on the has_suggestion field, other than that all the other controllers should be tested on normal threads that don't have suggestions. Does this help? Or would you like to discuss this further?

This might also help with some of the above comments, so I'll wait till this comment is addressed before doing a next review.

Reply:
We need has_suggestion=True as we are testing suggestion_thread_dicts. For example in ThreadListHandlerForTopicsHandler, you can see:
```
'suggestion_thread_dicts': (
                [t.to_dict() for t in feedback_services.get_all_threads(
                    feconf.ENTITY_TYPE_TOPIC, topic_id, True)])
            })
```
`get_all_threads` takes True as its last argument which stands for suggestion. The handlers I have tested they are for suggestions.

Reply:
Yeah. Are all of these tests only testing with suggestion_thread_dicts? I would suggest doing one test to make sure this is returned correctly, and all other tests should focus on feedback threads.

Reply:
Yes I think all of the tests are testing suggestion_thread_dicts, that's why has_suggestion=True is required.

Reply:
Just to clarify... Feedback threads might exist independently (without having suggestions). All the core functionality of feedback threads should be tested with such threads (which are most of the tests in this PR). I don't want the core functionality of feedback threads to be dependant on the presence of suggestions to be tested. I could setup a quick chat sometime if you want to clear up any confusions.

Thanks!

Reply:
Done!